### PR TITLE
feat: ✨ Mod directory creation from template

### PR DESCRIPTION
**Pull request by KANAjetzt**
- New template directory `res://addons/mod_tool/templates`
- New template `res://addons/mod_tool/templates/default`
- Added a *"Select Template"* Button to the Export tab
- Added a FileDialog to select a template directory
- Updated *create_mod.gd*
   - Copy the selected template dir
   - Rename it to the Mod-ID
   - Open the *mod_main.gd* file in the script editor
   - Update the *manifest.json* with the mod *name* and *namespace*
   - Open the manifest editor tab

### CODE REVIEW

File: addons/mod_tool/global/store.gd [hunk 1]
```diff
@@ -15,13 +15,17 @@ var editor_file_system: EditorFileSystem
 var error_color := ""
 
 var name_mod_dir := "" setget set_name_mod_dir
+var mod_name := ""
+var mod_namespace := ""
```
Comment (Qubus0) [inline addons/mod_tool/global/store.gd:19]: shouldn't these two be covered by the manifest?

File: addons/mod_tool/interface/create_mod/create_mod.gd [hunk 2]
```diff
@@ -18,31 +18,58 @@ func add_mod() -> void:
...
+		ModToolStore.editor_plugin.get_editor_interface().set_main_screen_editor("Script")
+
+		# Update the mod name in the manifest
+		ModToolStore.manifest_data.name = ModToolStore.mod_name
+
+		# Update the namespace in the manifest
+		ModToolStore.manifest_data.namespace = ModToolStore.mod_namespace
+
+		# Update manifest_dict
+		ModToolStore.manifest_dict = ModToolStore.manifest_data.get_as_dict()
```
[the remark this answers is not in the document]
Comment (KANAjetzt) [reply addons/mod_tool/interface/create_mod/create_mod.gd:62]: I would like to.
The dictionary is only used in the manifest editor in `update_ui()`.
If we can swap this line
`if ModToolStore.manifest_dict.has(input.key):`
with a check if the prop on the manifest exists, we can get rid of the dict.

https://github.com/GodotModding/godot-mod-tool/blob/d00787a0c0a00516fd7a0b41eb09226cc973c0d3/addons/mod_tool/interface/manifest_editor/manifest_editor.gd#L35-L47

File: addons/mod_tool/global/store.gd (hunk 1 above)
Comment (Qubus0) [reply addons/mod_tool/global/store.gd:19]: yeah that could work

Comment (KANAjetzt) [reply addons/mod_tool/global/store.gd]: *fixed [8afb088](https://github.com/GodotModding/godot-mod-tool/pull/45/commits/8afb088ca1026a3afa21bda3b09c538d5c55c117)*

**Review by Qubus0 — APPROVED** [review]

**Pull request merged**